lib/ddtrace/profiling/scheduler.rb

```diff
@@ -5,31 +5,39 @@
 
 module Datadog
   module Profiling
-    # Periodically (every DEFAULT_INTERVAL seconds) takes data from the `Recorder` and pushes them to all configured
+    # Periodically (every DEFAULT_INTERVAL_SECONDS) takes data from the `Recorder` and pushes them to all configured
     # `Exporter`s. Runs on its own background thread.
     class Scheduler < Worker
       include Workers::Polling
 
-      DEFAULT_INTERVAL = 60
-      MIN_INTERVAL = 0
+      DEFAULT_INTERVAL_SECONDS = 60
+      MIN_INTERVAL_SECONDS = 0
 
       attr_reader \
         :exporters,
         :recorder
 
-      def initialize(recorder, exporters, options = {})
+      def initialize(
+        recorder,
+        exporters,
+        # Should we flush immediately on the next call to flush_events, or loop/sleep at least once before doing it?
+        skip_next_flush: true,
```

Comment:
Do you think this should be configurable?
I'm not sure I understand a scenario where you'd want to not skip the first flush (except maybe for testing).

Reply:
Good question: No, I don't think it's needed.

I decided to go this route because there were already a number of tests that exercised `flush_events` directly and that would need to get a bit uglier if they couldn't disable this (e.g. the tests where we're using `false` for this).

To be honest, probably the best solution would even be to modify the `Datadog::Workers::IntervalLoop` to optionally have this behavior of "perform the wait once before trying to run the workload", since that's where it's really implemented.

I'll try to take a stab at it and see how it looks.

Reply:
I've tried this alternative in #1479 . As I commented there, I think both approaches are reasonable and solve the issue, although having done #1479, I think I like it a bit more than #1462. On the other hand, #1462 only modifies the one specific profiling class where we need this behavior, whereas #1479 modifies a core class to add a behavior that effectively has a single customer right now.

Let me know what you think :)